src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -323,26 +323,37 @@ private void LogPrimaryKeys()
             ColumnDefinition column;
             foreach ((string entityName, Entity _) in _entities)
             {
-                SourceDefinition sourceDefinition = GetSourceDefinition(entityName);
-                _logger.LogDebug("Logging primary key information for entity: {entityName}.", entityName);
-                foreach (string pK in sourceDefinition.PrimaryKey)
+                try
                 {
-                    column = sourceDefinition.Columns[pK];
-                    if (TryGetExposedColumnName(entityName, pK, out string? exposedPKeyName))
+                    SourceDefinition sourceDefinition = GetSourceDefinition(entityName);
```

Comment:
Can this call to `GetSourceDefinition` throw an exception, and if so, does the message about logging primary keys make sense?

If it can't throw an exception, does that mean that the only call in here that can is the call to TryGetExposedColumnName? If so, would it help to tailor the exception messaging toward that?

Reply:
`GetSourceDefinition` can throw exception. The message about logging primary keys still makes sense, because that is what we are trying to achieve and failure to do so should be logged with appropriate message. So if exception is thrown at `GetSourceDefinition`, it will give failure to  log primary keys because of SourceDefinition error.